integration-tests/minio/src/main/java/org/apache/camel/quarkus/component/minio/it/MinioResource.java

```diff
@@ -30,76 +34,169 @@
 import jakarta.ws.rs.GET;
 import jakarta.ws.rs.POST;
 import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
 import jakarta.ws.rs.Produces;
 import jakarta.ws.rs.QueryParam;
 import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import org.apache.camel.CamelContext;
 import org.apache.camel.ConsumerTemplate;
+import org.apache.camel.Exchange;
 import org.apache.camel.ProducerTemplate;
+import org.apache.camel.component.minio.MinioComponent;
 import org.apache.camel.component.minio.MinioConstants;
 import org.apache.camel.component.minio.MinioOperations;
+import org.apache.camel.util.Pair;
 
 @Path("/minio")
 @ApplicationScoped
 public class MinioResource {
 
     public static final String SERVER_ACCESS_KEY = "testAccessKey";
     public static final String SERVER_SECRET_KEY = "testSecretKey";
+    private static final String URL_AUTH = "accessKey=" + SERVER_ACCESS_KEY + "&secretKey=RAW(" + SERVER_SECRET_KEY + ")";
 
     @Inject
     ProducerTemplate producerTemplate;
 
     @Inject
     ConsumerTemplate consumerTemplate;
 
+    @Inject
+    CamelContext camelContext;
+
+    @Path("/consumerWithClientCreation/{endpoint}")
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response consumerWithClientCreation(@PathParam("endpoint") String endpoint) {
+        MinioComponent minioComponent = camelContext.getComponent("minio", MinioComponent.class);
+        MinioClient client = minioComponent.getConfiguration().getMinioClient();
+        minioComponent.getConfiguration().setMinioClient(null);
```

Comment:
May be better to have a dedicated component instance for this, then you don't need to mess with the original `minio` instance.

E.g

```
@Produces
@Named("minioCamel")
MinioComponent camelMinioComponent() {
    // Code to set null client, autowired disabled etc
}
```

Reply:
you are right, that would be "cleaner" solution, thanks for the hint, I'll change that

Reply:
done